compiler/rustc_lint/src/lints.rs

```diff
@@ -224,9 +225,39 @@ pub(crate) struct BuiltinConstNoMangle {
     pub suggestion: Span,
 }
 
+// This would be more convenient as `from: String` and `to: String`, but then we'll ICE for formatting a `Ty`
+// when the lint is `#[allow]`ed.
+pub(crate) struct TransmuteBreadcrumbs<'a> {
+    pub before_ty: &'static str,
+    pub ty: Ty<'a>,
+    pub after_ty: String,
+}
+
+impl IntoDiagArg for TransmuteBreadcrumbs<'_> {
+    fn into_diag_arg(self) -> DiagArgValue {
+        format!("{}{}{}", self.before_ty, self.ty, self.after_ty).into_diag_arg()
+    }
+}
+
+// mutable_transmutes.rs
 #[derive(LintDiagnostic)]
 #[diag(lint_builtin_mutable_transmutes)]
-pub(crate) struct BuiltinMutablesTransmutes;
+#[note]
+pub(crate) struct BuiltinMutablesTransmutes<'a> {
```

Comment:
Why does this one have a `Builtin` prefix but `UnsafeCellTransmutes` does not?

Reply:
I'm not sure what the  `Builtin` prefix means (some lints include it while some don't), so I just randomly chose not to include it for the new lint, but not change the existing one.